modules/openapi-generator/src/main/java/org/openapitools/codegen/config/CodegenConfigurator.java

```diff
@@ -643,6 +654,38 @@ public Context<?> toContext() {
         return new Context<>(specification, generatorSettings, workflowSettings);
     }
 
+    /*
+     * ODI-41 : Fix usage of x-protobuf-index for $ref fields
+     * This function automatically detect $ref that aren't part of an allOf and turn them into an allOf pattern
+     */
+    private void updateInputSpecReference() {
+        try {
+            String fileContent = new String(Files.readAllBytes(Paths.get(this.inputSpec)));
+            Pattern pattern = Pattern.compile("\\n\\s*(?<!-\\s?)\\$ref\\s?:.*\\n(?![\\s\\S]*components:)");
+            Matcher matcher = pattern.matcher(fileContent);
+
+            // Check if there's any $ref first
+            if (matcher.find()) {
+                File newInputSpec = new File("src/main/resources/codegen/temporaryInputSpec.yaml");
+                StringBuffer sb = new StringBuffer();
+                do {
+                    int location = matcher.group().indexOf("$");
+                    String replacement = matcher.group().substring(0, location) + "allOf:" + matcher.group().substring(0, location) + " - \\" + matcher.group().substring(location);
```

Comment:
Consider usage of SnakeYAML library to avoid indentation management

Reply:
Done